File: src/types/Appointment.ts
// Matches the backend's AppointmentStatus enum (BeautyCareClinic.Domain.Enums.AppointmentStatus).
// Phase 011 MVP only ever produces 'Scheduled' or 'Cancelled' via the API (Q2) — 'Completed'/
// 'NoShow' are kept in the union for forward-compatibility with the enum, but the frontend has no
// UI to set them in this phase.
export type AppointmentStatus = 'Scheduled' | 'Completed' | 'Cancelled' | 'NoShow';

// Aligned field-for-field with the backend's AppointmentDto
// (backend/BeautyCareClinic.Application/DTOs/AppointmentDtos.cs). Phase 011 Q1: `therapistId` was
// renamed to `userId` to match the backend and every other Phase 007-010 integration.
export interface Appointment {
  id: string;
  customerId: string;
  treatmentTypeId: string;
  treatmentTypeName: string;
  userId: string;
  userFullName: string;
  startTime: string;  // naive local ISO: YYYY-MM-DDTHH:mm:ss
  endTime: string;    // naive local ISO: YYYY-MM-DDTHH:mm:ss
  status: AppointmentStatus;
  createdAt: string;  // naive local ISO
}

File: backend/BeautyCareClinic.Domain/Entities/Appointment.cs
using BeautyCareClinic.Domain.Enums;

namespace BeautyCareClinic.Domain.Entities;

public class Appointment
{
    public Guid Id { get; set; }
    public Guid CustomerId { get; set; }
    public Guid TreatmentTypeId { get; set; }
    public Guid UserId { get; set; }
    public DateTime StartTime { get; set; }
    public DateTime EndTime { get; set; }
    public AppointmentStatus Status { get; set; } = AppointmentStatus.Scheduled;
    public DateTime CreatedAt { get; set; }

    /// <summary>Snapshotted at creation/reschedule time, matching the Treatment.PerformedByFullName /
    /// Note.WrittenByFullName / Payment.RecordedByFullName pattern.</summary>
    public string UserFullName { get; set; } = string.Empty;

    public Customer Customer { get; set; } = null!;
    public TreatmentType TreatmentType { get; set; } = null!;
    public User User { get; set; } = null!;
}
